### PR TITLE
cmd: restore-repo --units is repeated

### PR DESCRIPTION
**Pull request by realaravinth**
It is not a coma separated value. Add tests and fix the documentation.

---
This PR originates from [@dachary](https://lab.forgefriends.org/dachary) of the [forgefriends](https://forgefriends.org) project. [Please see here for origin](https://lab.forgefriends.org/forgefriends/forgefriends/-/merge_requests/51).

### CODE REVIEW

**Review by Gusted — APPROVED** [review]

**Review by lunny — APPROVED** [review]

File: cmd/restore_repo.go [hunk 1]
```diff
@@ -15,6 +15,17 @@ import (
...
+	"wiki",
+	"issues",
+	"labels",
+	"releases",
+	"release_assets",
+	"milestones",
+	"pull_requests",
+	"comments",
+}
+
```
Comment (wxiaoguang) [inline cmd/restore_repo.go:28]: I think this default value is not ideal for the server code.

The default value should be 0-length slice to indicate all units (there may be more in the future).

update: if it guarantees that the `units` always contains the units to be restored, the support for `len(units) == 0` should be deleted to make the logic consistent.

image

File: cmd/restore_repo.go [hunk 2]
```diff
@@ -37,11 +48,10 @@ var CmdRestoreRepository = cli.Command{
...
 			Usage: "Restore destination repository name",
 		},
-		cli.StringFlag{
+		cli.StringSliceFlag{
 			Name:  "units",
-			Value: "",
-			Usage: `Which items will be restored, one or more units should be separated as comma.
-wiki, issues, labels, releases, release_assets, milestones, pull_requests, comments are allowed. Empty means all units.`,
+			Value: &defaultUnits,
+			Usage: "Which items will be restored, can be repeated",
```
Comment (wxiaoguang) [inline cmd/restore_repo.go:54]:
```suggestion
			Usage: "Which items (wiki, issues, labels, releases, release_assets, milestones, pull_requests, comments) will be restored, can be repeated. Not providing this flag means all units.",
```

**Comment by wxiaoguang** [comment]
Since it works, no blocker from my side. Just put my thoughts here.

**Review by delvh — APPROVED** [review]

File: cmd/restore_repo.go (hunk 1 above)
Comment (lunny) [reply cmd/restore_repo.go:28]: I think it's a good idea to reduce the possibility to forget the places when we support more units.

**Comment by realaravinth** [comment]
This pull request is closed as [requested by the author](https://lab.forgefriends.org/forgefriends/forgefriends/-/merge_requests/51#note_1392) who will not be able to answer further questions here because they do not have a GitHub account.

**Pull request closed without merging**